planning/behavior_path_planner/src/scene_module/side_shift/side_shift_module.cpp

```diff
@@ -182,14 +198,18 @@ bool SideShiftModule::addShiftPoint()
   };
 
   // remove shift points on a far position.
-  for (int i = static_cast<int>(shift_points.size()) - 1; i >= 0; --i) {
-    const auto dist_to_start = calcLongitudinal(shift_points.at(i));
-    const double remove_threshold =
-      std::max(planner_data_->self_odometry->twist.twist.linear.x * 1.0 /* sec */, 2.0 /* m */);
-    if (dist_to_start > remove_threshold) {  // TODO(Horibe)
-      shift_points.erase(shift_points.begin() + i);
-    }
-  }
+  const auto remove_iter = std::remove_if(
+    shift_points.begin(), shift_points.end(), [this, calcLongitudinal](const ShiftPoint & sp) {
+      const auto dist_to_start = calcLongitudinal(sp);
+      constexpr double max_remove_threshold_time = 1.0;  // [s]
+      constexpr double max_remove_threshold_dist = 2.0;  // [m]
```

Comment:
Is this change separate from this bug?

Reply:
The change does not directly related to the bug.|
However, it very slightly increase the performance and it should allow the shift point above threshold to be remove more "smoothly", which may result in more stable path.

Reply:
However, removing this changes will not affect how the bug is fixed, therefore, I can always revert this part